packages/rol/adapters/mpi/src/function/ROL_PinTHierarchy.hpp

```diff
@@ -603,7 +603,7 @@ class PinTHierarchy {
          MPI_Recv(&sizeA,1,MPI_INT,2*rank  ,level,mpiComm,MPI_STATUS_IGNORE);
          MPI_Recv(&sizeB,1,MPI_INT,2*rank+1,level,mpiComm,MPI_STATUS_IGNORE);
 
-         TEUCHOS_ASSERT(sizeA+sizeB==2*stamps_[level]->size());
+         TEUCHOS_ASSERT(sizeA+sizeB==2*(long long)stamps_[level]->size());
```

Comment:
That is a C-style cast.  Might consider `static_cast<long long>(...)`?

Reply:
`long long (value)` is also legit.

Reply:
> `long long (value)` is also legit.

Unfortunately, I believe that that is equivalent to `(long long) value` which is a C-style cast.

A correct implicit conversion in C++11 is `long long {value}` I think.  If the conversion from `value` to `long long` is not allowed it will issue an error.  If it is not safe, it will issue a warning.